lib/internal/modules/esm/hooks.js

```diff
@@ -35,6 +35,7 @@ const {
 } = require('internal/errors').codes;
 const { exitCodes: { kUnfinishedTopLevelAwait } } = internalBinding('errors');
 const { URL } = require('internal/url');
+const { canParse: urlCanParse } = internalBinding('url');
```

Comment:
nit: this sounds a little yoga-esque

```suggestion
const { canParse: canParseURL } = internalBinding('url');
```

Reply:
Not sure this is better, `isURLString` might be more suited. `URLCanParse` has the upside of being consistent with how primordials are named, which is nice imho

Reply:
I want to change this to URLCanParse too but I'm too demotivated by having to run Node.js CI multiple times because of the flaky tests.